### PR TITLE
Improve cluster state tracking

### PR DESCRIPTION
**Pull request by cket**
Work in progress, opening to get an idea of what tests I broke

### CODE REVIEW

**Comment by cket** [comment]
Jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
@ejacox this should be ready to go!

File: src/toil/provisioners/abstractProvisioner.py [hunk 1]
```diff
@@ -52,7 +46,7 @@ class AbstractProvisioner(object):
     __metaclass__ = ABCMeta
 
 
-    def __init__(self, config=None, batchSystem=None):
+    def __init__(self, config=None):
         """
```
Comment (jessebrennan) [inline src/toil/provisioners/abstractProvisioner.py]: Why doesn't this take in batchSystem as an argument anymore? Also the docstring below needs to be changed to reflect that. I realize this has already been merged, but I'm hoping @cket can help explain this design choice.

Comment (cket) [inline src/toil/provisioners/abstractProvisioner.py]: The batch system was only referenced in the higher level logic that I moved into the cluster scaler class so there was no need to keep it in this class as well. But yeah the doc string should be updated

**Pull request merged**